packages/spright-components/src/chat/specs/README.md

```diff
@@ -0,0 +1,252 @@
+
+The message component will allow slotting arbitrary content, but any efforts to add content types to Nimble are out of scope of this document. For example, adding capabilities to the rich text viewer or adding styling for specific content types.
+
+### Features
+
+#### Chat message
+
+1. Display arbitrary slotted content. For example: text, rich text, buttons, or a spinner.
+1. Layout content to the right, center, or left of parent container depending on metadata about who sent the message.
+1. Size based on content size with maximum width (but not height) based on parent's width.
+1. Change the styling of the message depending on metadata about who sent the message. For example: render user messages in a bubble with the tail pointing to the right but render system messages with no styling.
+
+#### Chat conversation
+
+1. Lays out messages vertically based on their order.
+1. Displays a vertical scrollbar if there are more messages than fit in the height allocated to the conversation.
+1. Only appearance of its own is to set a background color.
+
+### Risks and Challenges
+
+These components are competing against possible implementations within applications. Depending on who implements these components, the overhead of learning the Nimble repo's tech stack could introduce a small risk.
+
+### Prior Art/Examples
+
+**Screenshot of Figma design of chat and conversation component (light mode)**
+![ ](spec-images/chat-conversation.png)
+
+**Screenshot of Figma design of chat components embeded within larger pane (dark mode)**
+![ ](spec-images/chat-pane.png)
+
+---
+
+## Design
+
+### Examples
+
+#### Text conversation example
+
+```html
+<spright-chat-conversation>
+    <spright-chat-message status="incoming">
+        Hi, how can I help?
+    </spright-chat-message>
+    <spright-chat-message status="outgoing">
+        I need to analyze my data to find anomalies.
+    </spright-chat-message>
+    <spright-chat-message status="system">
+        <nimble-spinner></nimble-spinner>
+    </spright-chat-message>
+</spright-chat-conversation>
+```
+
+#### Rich text message example
+
+```html
+<spright-chat-conversation>
+    <spright-chat-message status="incoming">
+        <nimble-rich-text-viewer id="welcome"></nimble-rich-text-viewer>
+    </spright-chat-message>
+</spright-chat-conversation>
+```
+
+```js
+const richText = document.querySelector('#welcome');
+richText.markdown = 'Welcome **Homer**, how can I help?';
+```
+
+#### Prompt buttons message example
+
+```html
+<spright-chat-message status="system">
+    <nimble-button appearance="block">Help with my taxes</nimble-button>
+    <nimble-button appearance="block">Provide me some life advice</nimble-button>
+</spright-chat-message
+```
+
+### API
+
+#### Message
+
+- _Component Name_ `spright-chat-message`
+- _Props/Attrs_
+    - `status = "incoming" | "outgoing" | "system"`
```

Comment:
Any preference of incoming/outgoing vs inbound/outbound? Seems like the latter is used more for tech / messaging (email inbox, sms, etc). Looks like the terms used in lightning as well (didn't search more broadly): https://www.lightningdesignsystem.com/components/chat/

Reply:
I'm good with inbound/outbound over incoming/outgoing. Now that I think of it the "going" options sound like it's still in the process of being sent. Some of the options in the other thread might change this so let's keep brainstorming there.